### ISSUE TITLE
Tokenizer breaks on empty strings, carriage returns ... 

### ISSUE THREAD

**Issue by nitishgupta** [CogComp/cogcomp-nlp#278]
I experienced that our latest tokenizer breaks on empty string "" and  "\n" at the least.
It produces an 'ArrayOutOfBound Exception'

**Comment by danyaljj** [comment]
So as you can see in #282, I added tests which contain empty string and new line, and the tests pass. 
So unless you give an explicit way to reproduce the exception, we won't be able to solver this.

**Comment by nitishgupta** [comment]
From #282 I learned that creating TextAnnotation using TextAnnotationBuilder fails on string that are empty, just carriage return, space etc.
Code snippet that fails and gives 'java.lang.ArrayIndexOutOfBoundsException: -1' : 
`
String text = " "; // or "\n" or ""
TextAnnotationBuilder tab = new TokenizerTextAnnotationBuilder(new IllinoisTokenizer());
TextAnnotation ta = tab.createTextAnnotation(text);
`

**Comment by nitishgupta** [comment]
Thanks Daniel and Mark!

**Comment by danyaljj** [comment]
Re-opening the issue since apparently it still fails when receiving carriage returns (according to @nitishgupta )

**Comment by nitishgupta** [comment]
This fails when I do this in Python. I haven't checked in Java recently. Could ccg_nlpy be using an old version?

**Comment by danyaljj** [comment]
It is using a slightly older version; but very recent: https://github.com/CogComp/cogcomp-nlpy/blob/9e733c0d936cbd923a9e6702e36bc6d64ab887ad/ccg_nlpy/download.py#L14 

Is there a way to verify that it is a purely a Java issue (and Python side is not to blame)?

**Comment by nitishgupta** [comment]
I can do that and update here.

**Comment by nitishgupta** [comment]
```
TextAnnotationBuilder tab = new TokenizerTextAnnotationBuilder(new StatefulTokenizer());
tab.createTextAnnotation("", "", "\n");
tab.createTextAnnotation("", "", " ");
```
-- both break with the latest version.

**Comment by mssammon** [comment]
I don't think a TextAnnotation object should be created for such cases, because no annotations can be generated for them (at least, not with our current tooling).  However, the failure should result in an explicit exception to this effect rather than resulting in a bounds exception.

**Comment by mssammon** [comment]
@nitishgupta  what is the use case? would a TextAnnotation with no non-whitespace text be useful in some way?

**Comment by nitishgupta** [comment]
Having a TextAnnotation for these edge cases reduces the burden on the user
to write checking-code.
IMO, a TextAnnotation should be able to be created for any String object.
Is that not the general consensus?

**Comment by mssammon** [comment]
I disagree. User of the resulting TextAnnotation will likely have to check for empty views -- or, the unlucky client who is using their code as an intermediary will. How is this better?

**Comment by nitishgupta** [comment]
Okay. I get your point. Thanks for the time.
